src/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -1339,6 +1437,51 @@ sed -e "s|^root:[^\:]*:|root:${root_shadow}:|" /etc/shadow > /usbkey/shadow \
 
 cp -rp /etc/ssh /usbkey/ssh || fatal "failed to set up preserve host keys"
 
+if [ $ondisk == "yes" ]; then
+    printf "%-56s" "Creating self-bootable $BOOTPOOL pool... "
+
+    # Reality check $BOOTPOOL was created with -B.
+    # Easiest way to do this is to check for the `bootsize` property not
+    # its default, which is NO bootsize.
+    zpool get bootsize ${BOOTPOOL} | grep -q -w default
+    if [[ $? == 0 ]]; then
+	fatal "\nDesired bootable pool $BOOTPOOL was not created with -B"
+    fi
+
+    # Create BOOTPOOL/boot and set bootfs.
+    zfs create ${BOOTPOOL}/boot || fatal "Cannot create boot filesystem"
+    zpool set bootfs=${BOOTPOOL}/boot ${BOOTPOOL}
+    if [[ $? != 0 ]]; then
+	fatal "\nCannot set bootfs"
+    else
+	# Get "install media mounted" and copy over boot stuff:
+	copy_installmedia
+
+	# Append 'fstype="ufs"' to /${BOOTPOOL}/boot/boot/loader.conf for
+	# ramdisk root.
+	echo 'fstype="ufs"' >> /${BOOTPOOL}/boot/boot/loader.conf || \
+	    fatal "Can't append to /${BOOTPOOL}/boot/boot/loader.conf"
+
+	# Determine which disk(s) to install things in.
+	# Then for each disk:
+	# 	installboot -m -b....
+	#
+	# We created the pool, so we can assume
+	# `zpool create -B` generates things such that for disk X, Xs0 == ESP,
+	# Xs1 == data-for-BOOTPOOL
+	for a in \
+	`zpool list -v ${BOOTPOOL} | egrep 'c[0-9]+' | awk '{print $1}'`; do
```

Comment:
This is probably safe in this context, but in general somewhat dangerous.

Prefer:

```
mapfile -t boot_devices < <(zpool list -v "${BOOTPOOL}" | grep -E 'c[0-9]+' | awk '{print $1}')
for a in "${boot_devices[@]}"; do
```

Also, should the regex be more specific? `c[0-9A-F]+t[0-9A-F]+d[0-9A-F]+'`? (I don't know if the `c` or `d` values can go up to A-F, but `t` can.)

Reply:
I don't see `mapfile` readily available for the SmartOS installer.  (It's not in smartos-live/src either.)

As for the regex, old-school IDE (like on some VM/cloud environments) have cNdN.  I figure matching on c{num} was okay, but if the controller counts in hex, yeah, we need A-F too.

Reply:
It’s a bash shell builtin in v4, so will be available.

Reply:
Wow.  I thought bash would answer "shell built-in" to `which mapfile`.  :(  Thanks.  Will update later today.

Reply:
Check now.

Reply:
Now all in piadm.